test/parallel/test-http-client-abort3.js

```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+
+const req = http.get('http://[2604:1380:45f1:3f00::1]:4002');
```

Comment:
I think we should not rely on this specific IP/port. We could use a custom agent with a custom `createConnection()` that creates a socket and destroys it with an error on next tick. This would also allow us to use an error that is not OS dependent.

Reply:
We should also test the no agent case.

Reply:
If I am not mistaken, to test the 'no agent case' a `createConnection` function has to be passed. That one maybe can be reused in the custom Agent. The test on the os-dependant error could be less strict (i.e. test multiple codes) to avoid fails in different platforms.

Reply:
Yes, it could be something like this:

```js
'use strict';

const common = require('../common');
const http = require('http');
const net = require('net');

function createConnection() {
  const socket = new net.Socket();

  process.nextTick(function() {
    socket.destroy(new Error('Oops'));
  });

  return socket;
}

{
  const req = http.get({ createConnection });

  req.on('error', common.expectsError({ name: 'Error', message: 'Oops' }));
  req.abort();
}

{
  class CustomAgent extends http.Agent {}
  CustomAgent.prototype.createConnection = createConnection;

  const req = http.get({ agent: new CustomAgent() });

  req.on('error', common.expectsError({ name: 'Error', message: 'Oops' }));
  req.abort();
}
```

@ronag feel free to use it.